typed-builder-macro/src/struct_info.rs

```diff
@@ -462,10 +462,30 @@ impl<'a> StructInfo<'a> {
             let name = &field.name;
             if let Some(ref default) = field.builder_attr.default {
                 if field.builder_attr.setter.skip.is_some() {
-                    quote!(let #name = #default;)
+                    if field.builder_attr.mutable_during_default_resolution {
```

Comment:
Instead of adding the mutability each time, you can add it afterwards:

```rust
let field_assignment = if let Some(ref default) = field.builder_attr.default {
    // Original code
} else {
    // This is the same as the original:
     quote!(let #name = #name.0;)
};
if field.builder_attr.mutable_during_default_resolution {
    quote! {
        #field_assignment
        let mut #name = #name;
    }
} else {
    field_assignment
}
```

Or maybe:
```rust
let field_value = if let Some(ref default) = field.builder_attr.default {
    // Original code, but without the `let #name =` and the `;` in the quotes
} else {
    // Like this:
     quote!(#name.0)
};
if field.builder_attr.mutable_during_default_resolution {
    quote!(let mut #name = #field_value;)
} else {
    quote!(let #name = #field_value;)
}
```

Or even better:

```rust
let maybe_mut = if field.builder_attr.mutable_during_default_resolution {
    quote!(mut)
} else {
    quote!()
};
if let Some(ref default) = field.builder_attr.default {
    // Original code, but use `#maybe_mut` to set the mutability
} else {
    // Like this:
     quote!(let #maybe_mut #name = #name.0;)
}
```

I think this method is preferable because we can then later add a span to `maybe_mut` and get better warnings.